.github/workflows/dev_deployment.yml

```diff
@@ -64,11 +64,19 @@ jobs:
             echo "Build and push docker image with that tag $IMAGE_TAG "
         fi
     
-    - name: Build, tag, and push base image to Amazon ECR
-      id: build-base-image
+    - name: Base Image Changes
+      uses: dorny/paths-filter@v2
+      id: base-image-changes
+      with:
+        filters: |
+          django_base:
+           - ./docker/production/django_base/Dockerfile
       env:
         ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
         ECR_REPOSITORY: ${{ env.ECR_BASE_REPOSITORY_NAME }}
+
+    - name: Build, tag, and push base image to Amazon ECR
+      if: steps.base-image-changes.outputs.django_base == 'true'  
       run: |
         echo "base_image=$ECR_REGISTRY/$ECR_REPOSITORY:${{ steps.set-version-number.outputs.version }}" >> $GITHUB_OUTPUT
         docker build -t $ECR_REGISTRY/$ECR_REPOSITORY:${{ steps.set-version-number.outputs.version }} -f ./docker/production/django_base/Dockerfile .
```

Comment:
it will fail on line 91 if you do not build a image in this step - change that to use the latest image